sw.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-env serviceworker */
+self.addEventListener('install', function (event) {
+  self.skipWaiting()
+  sendMessage(event, 'render')
+})
+
+self.addEventListener('activate', function (event) {
+  sendMessage(event, 'render')
+})
+
+self.addEventListener('sync', function (event) {
+  sendMessage(event, 'render')
+  self.registration.showNotification(event.tag)
+})
+self.onmessage = function (e) {
+  if (e.data === 'ping') {
+    sendMessage(e, 'Message sent from the service worker')
+  } else {
+    self.registration.showNotification(e.data)
+  }
+}
+self.addEventListener('fetch', function (event) {
+  event.respondWith(async function () {
+    // Respond from the cache if we can
+    const cachedResponse = await caches.match(event.request)
```

Comment:
oh no, we definitely can't use `await` yet! Some browsers (e.g. safari) might support service workers, but not quite yet async / await

Reply:
Ok, I'll change that. Anyway the `sw.js` file is not included in the npm package, it is just for the example.

Reply:
Also, [support for fetch](http://caniuse.com/#search=fetch) API is similar to the [support for async await](http://caniuse.com/#search=async) (In fact, safari wouldn't be affected)

Reply:
Ok, promises are confusing :disappointed:

```js
event.respondWith(function () {
    // Respond from the cache if we can
    caches.match(event.request).then(function (cachedResponse) {
      if (cachedResponse) return cachedResponse
      // Use the preloaded response, if it's there
      else return event.preloadResponse
    }).then(function (response) {
      if (response) return response
      // Else try the network.
      return fetch(event.request) /* global fetch */
    })
  }())
```

Is this the way the code should look with promises? If it's ok, I'll commit, otherwise I need a hand :frowning:

Reply:
Did a quick edit, to remove the extra  function call, but I think it's great otherwise! :D If you've tested it, we should merge!

```js
event.respondWith(caches.match(event.request).then(function (cachedResponse) {
      if (cachedResponse) return cachedResponse
      // Use the preloaded response, if it's there
      else return event.preloadResponse
    }).then(function (response) {
      if (response) return response
      // Else try the network.
      return fetch(event.request) /* global fetch */
    }))
```